File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Ruby/execute_javascript_test.rb
require 'rubygems'
require 'selenium-webdriver'
require 'test/unit'

class ExecuteJavaScriptTest < Test::Unit::TestCase
	def setup
		@driver = Selenium::WebDriver.for :chrome
		@driver.get 'http://www.google.com'
		@verification_errors = []
	end
	
	def test_javascript_calls
		title = driver.execute_script("return document.title").to_s();
		assert_equal "Google", title
		
		links = @driver.execute_script("var links = document.getElementsByTagName('A'); return links.length").to_s()
		assert_equal "42", links
	end

	def teardown
		@driver.quit
		assert_equal [], @verification_errors
	end
end

File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Python/execute_javascript_test.py
from selenium import webdriver
from array import *
import time, unittest

class ExecuteJavaScriptTest (unittest.TestCase):
	def setUp(self)	:
		self.driver = webdriver.Chrome()
		
	def test_execute_java_script(self):
		driver = self.driver
		driver.get("http://www.google.com")
		
		title = driver.execute_script("return document.title")
		self.assertEquals("Google",title)
		
		links = driver.execute_script("var links = document.getElementsByTagName('A'); return links.length")
		self.assertEquals(42, links)
		
	def tearDown(self):
		self.driver.close()
		
if __name__ == "__main__":
	unittest.main()